jcontroller.php

```diff
@@ -173,6 +178,34 @@ protected function refRequest($param)
         return $this->refRequest[$param];
     }
 
+    /**
+     * Filters access to an action based on the controller methods.
+     *  
+     * If access is granted, it proceeds to display the action.
+     * If access is not granted, it displays an error code (401 - Access Forbidden).
+     * 
+     * @return void
+     */
+    public function filterAccessControl()
+    {
+        if (!isset($_GET['action'])) {
+            return;
+        }
+
+        $actionName = $_GET['action'];
+        $accessRules = $this->accessRules();
+
+        if (!array_key_exists($actionName, $accessRules)) {
+            return;
+        }
+
+        $accessMethod = $accessRules[$actionName];
+
+        if ($accessMethod != '' && !$this->$accessMethod) {
```

Comment:
Your code allows access method only to be accessed through `$this` and it does not take any parameters.  We may also want to use methods from other classes and use parameters. For example, we may want to use something like this for access validation, 

```
User::canAccessModule('admin', 'user-management')
```

So, the your `accessRule` array may look like this,

```
[
     // calls User::canAccess('admin, 'user-management' 
    'actionA' => ['User::canAccess', ['admin', 'user-management']], 
    // calls $this->canAccessActionB
    'actionB' => [[$this, 'canAccessActionB']]
] 
```